### PR TITLE
luaPackages.lsqlite3complete: init at 0.9.5-1

### PR DESCRIPTION
**Pull request by spencerpogo**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: maintainers/scripts/luarocks-packages.csv [hunk 1]
```diff
@@ -34,6 +34,7 @@ lpty,,,,,,
 lrexlib-gnu,,,,,,
 lrexlib-pcre,,,,,,vyp
 lrexlib-posix,,,,,,
+lsqlite3complete,,,,,,
```
Comment (teto) [inline maintainers/scripts/luarocks-packages.csv:37]: could you add yourself as maintainer here ? (last column) . Rerunning the update script will update the meta.maintainers key of lsqlite3complete but it's probably faster to update it manually here https://github.com/NixOS/nixpkgs/pull/185719/files#diff-5907dc7a16a18d3c7ce0d98e89c9c340d2fd2686e20a1592371bb1bb11b3ba89R1036

**Comment by teto** [comment]
nix can't unpack the src, looks like we should use fetchTarball here. As it looks like you are going to be involved in the lua ecosystem for a while would you mind fixing maintainers/scripts/update-luarocks-packages  so that it indeed uses fetchTarball here ? alternatively you could override the source pkgs/development/lua-modules/overrides.nix  but that's more a bandaid while the first solution would benefit everyone

**Comment by Scoder12** [comment]
Would that need a separate PR?

**Comment by Scoder12** [comment]
Might be cleanest to add this change to https://github.com/NixOS/nixpkgs/pull/185723

**Comment by teto** [comment]
> Might be cleanest to add this change to https://github.com/NixOS/nixpkgs/pull/185723

True so you will need to rebase this one once https://github.com/NixOS/nixpkgs/pull/185723 is merged

**Comment by Scoder12** [comment]
> would you mind fixing maintainers/scripts/update-luarocks-packages so that it indeed uses fetchTarball here?

sure, but that code is actually in https://github.com/nix-community/luarocks-nix

**Comment by Scoder12** [comment]
I don't know how to run luarocks from a local checkout

**Comment by Scoder12** [comment]
nvm, figured it out. For future reference:

Add to the top of `src/bin/luarocks`:
```lua
package.loaded["luarocks.core.hardcoded"] = { SYSCONFDIR = [[/nix/store/y6ck6hwqyl0a9731dmv6a3q3dxc0xl37-luarocks-3.9.0/etc/luarocks]] }
```
Run:
`LUA_PATH='./src/?.lua;./src/?/init.lua' lua src/bin/luarocks`

**Comment by teto** [comment]
the `LUA_PATH='./src/?.lua;./src/?/init.lua' lua src/bin/luarocks` kinda appears in the shellHook of pkgs/development/tools/misc/luarocks/default.nix

**Comment by teto** [comment]
> I don't know how to run luarocks from a local checkout 

nix develop nixpkgs#luarocks

**Comment by Scoder12** [comment]
Can this merge?

**Comment by teto** [comment]
this doesn't build so not yet ?

**Comment by Scoder12** [comment]
Result of `nixpkgs-review pr 185719` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)

**Comment by Scoder12** [comment]
Result of `nixpkgs-review pr 185719` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>lua51Packages.lsqlite3complete</li>
    <li>lua52Packages.lsqlite3complete</li>
    <li>lua53Packages.lsqlite3complete</li>
    <li>luajitPackages.lsqlite3complete</li>
  </ul>
</details>

**Comment by teto** [comment]
I preferred the previous version of this PR with the generated code. 
Ideally I would prefer to merge 
1. https://github.com/nix-community/luarocks-nix/pull/14
2.  https://github.com/NixOS/nixpkgs/pull/185723 with an updated luarocks-nix
3. this

If you are in a hurry, you can restore the previous version of this PR with the fetchurl -> fetchtarball change and we will try to merge the other PRs fast enough that the next lua package set bump doesnt revert the fetchTarball into fetchurl.

**Comment by Scoder12** [comment]
Wouldn't that still not build because of the glibc and pkg-config dependencies?

**Comment by teto** [comment]
> Wouldn't that still not build because of the glibc and pkg-config dependencies?

these can be added to pkgs/development/lua-modules/overrides.nix  like for other packages

**Comment by Scoder12** [comment]
@teto Doesn't seem to be working, the zip URL from sqlite.org is detected as a git URL and the fetching fails

Responsible code path: https://github.com/nix-community/luarocks-nix/blob/bf320edf3ff71492bdb6b1fbe6794a557c1f33cb/src/luarocks/cmd/nix.lua#L454-L459

**Comment by teto** [comment]
damn I introduced a regression ? I will try to edit the README to attract new contributors. There should be some tests too since it's so annoying to make releases for it. Feel free to fix it / do a release if you have time. I am dealing with some issues of my own :'( https://github.com/NixOS/nixpkgs/pull/190067

**Comment by Scoder12** [comment]
It doesn't seem to be a regression, checked https://github.com/nix-community/luarocks-nix/commit/a42c46075118680d425b85f7b3551c22efa41e8d
```
❯ git status
HEAD detached at a42c460
nothing to commit, working tree clean
❯ nix develop . -c luarocks nix 'http://lua.sqlite.org/index.cgi/zip/lsqlite3_fsl09y.zip'
Initialized empty Git repository in /run/user/1000/nix-shell.int8Ew/git-checkout-tmp-4KGRwATX/lsqlite3_fsl09y.zip/.git/
fatal: http://lua.sqlite.org/index.cgi/zip/lsqlite3_fsl09y.zip/info/refs not valid: could not determine hash algorithm; is this a git repository?
fatal: http://lua.sqlite.org/index.cgi/zip/lsqlite3_fsl09y.zip/info/refs not valid: could not determine hash algorithm; is this a git repository?
fatal: http://lua.sqlite.org/index.cgi/zip/lsqlite3_fsl09y.zip/info/refs not valid: could not determine hash algorithm; is this a git repository?
fatal: http://lua.sqlite.org/index.cgi/zip/lsqlite3_fsl09y.zip/info/refs not valid: could not determine hash algorithm; is this a git repository?
Unable to checkout HEAD from http://lua.sqlite.org/index.cgi/zip/lsqlite3_fsl09y.zip.
Call to nix-prefetch-git --fetch-submodules --quiet http://lua.sqlite.org/index.cgi/zip/lsqlite3_fsl09y.zip failed

Error: LuaRocks 3.9.1 bug (please report at https://github.com/luarocks/luarocks/issues).
Arch.: linux-x86_64
src/luarocks/cmd/nix.lua:442: attempt to concatenate local 'storePath' (a nil value)
stack traceback:
        src/luarocks/cmd/nix.lua:442: in function <src/luarocks/cmd/nix.lua:377>
        (tail call): ?
        [C]: in function 'xpcall'
        src/luarocks/cmd.lua:672: in function 'run_command'
        /home/spencer/github/luarocks-nix/src/bin/luarocks:36: in main chunk
        [C]: ?
```

**Comment by Scoder12** [comment]
Interesting thing, trying to fetch from any `git://github.com` URL just hangs indefinitely for me so I can't even test `nix-prefetch-git`, and I can't seem to find any info about it online.

**Comment by Scoder12** [comment]
@teto could you take a look at this?

**Comment by Scoder12** [comment]
It seems to pass any URL it gets into `gen_src_from_git_url`: https://github.com/nix-community/luarocks-nix/blob/7b3cc90ebf07813ed511f530cc50d602e9502001/src/luarocks/cmd/nix.lua#L459

Should there be a check around this code to use a different prefetcher if needed?

**Comment by Scoder12** [comment]
@teto what can I do to get this merged? I know it isn't acceptable currently, but I am not a Lua programmer, and I don't think I can fix luarocks-nix without help.

**Comment by teto** [comment]
the package does't seem very active so let's help contributors, we can convert it to an automated one once luarocks-nix improves.

**Comment by zowoq** [comment]
This broke eval, reverted in 2c4b97d6a0eb6beead204afd4e67c63ea1ad06a0.

**Comment by wegank** [comment]
@teto Would you like to create a new PR for the package?

**Comment by Scoder12** [comment]
Yeah, not sure why it was green. The solution was mentioned above: https://github.com/NixOS/nixpkgs/pull/185719#issuecomment-1212223926

**Pull request merged**